pytest_html/resources/main.js

```diff
@@ -190,3 +190,33 @@ function one_row_for_data() {
         }
     });
 }
+
+function filter_table(elem) {
+    var outcome_att = "data-test-result";
+    var results_table = document.getElementById("results-table");
+
+    var rows = results_table.rows; 
+    var outcome = elem.getAttribute(outcome_att);
+    var hidden = !elem.checked;
+
+    //It hides all the rows that match the outcome
+    for(var i = 0; i < rows.length; i++){
+        if(rows[i].getAttribute(outcome_att) == outcome) {
+            rows[i].hidden = hidden;
+        }
+    }
+
+    hide_extra_html_by_attribute(hidden, outcome_att, outcome);
+}
+
+function hide_extra_html_by_attribute(hidden, attribute_key, attribute_value) {
```

Comment:
@davehunt I don't think that combining those loops will help a lot, right now we have a complexity r + l where r is the size of rows and l the size of logs, in big O notation, it would be only O(r), so it is a algorithm with a linear complexity. Combining those loops, we still would have a linear  complexity (so speaking about only complexity, it would not improve the algorithm), but we would have some conditions:
=> There could be a bigger chance of accessing a index that don't exist
=> The loop would be confusing
=> The could would not be reusable
=> We would have to do many checks (such as to know which array is bigger, check if the index is not bigger than the array length, and maybe others, it smells like bug)
I split this methods in two because the second method can be used on the issue of collapsing tests logs, and also in the search issue (because we will want to collapse the logs of the methods that don't matches the search).